File: archive/c/muse/algorithms/search/LinearSearchTest.c
#include <muse/algorithms/search/LinearSearch.h>
#include <muse/util/SequenceBuilder.h>
#include <muse/util/TestRunner.h>

bool testLinearSearch(void) {
  size_t size = 32768;

  int arr[size];
  SequenceBuilder_packIncreasing(arr, size);

  if (LinearSearch_find(arr, size, -1) != size) {
    return false;
  }

  if (LinearSearch_find(arr, size, 2147483647) != size) {
    return false;
  }

  for (size_t i = 0; i < size; i++) {
    if (LinearSearch_find(arr, size, arr[i]) != i) {
      return false;
    }
  }

  return true;
}

int main(void) { TestRunner_pick(&testLinearSearch); }

File: archive/php/muse/algorithms/search/LinearSearchTest.php
<?php

declare(strict_types=1);

require_once "muse/algorithms/search/LinearSearch.php";
require_once "muse/util/SequenceBuilder.php";
require_once "muse/util/TestRunner.php";

use muse\algorithms\search\LinearSearch;
use muse\util\SequenceBuilder;
use muse\util\TestRunner;

function testLinearSearch(): bool
{
    $size = 32768;

    $arr = array_pad(array(), $size, 0);
    SequenceBuilder\packIncreasing($arr);

    if (LinearSearch\find($arr, -1) != $size) {
        return false;
    }

    if (LinearSearch\find($arr, 2_147_483_647) != $size) {
        return false;
    }

    foreach ($arr as $i => $v) {
        if (LinearSearch\find($arr, $v) != $i) {
            return false;
        }
    }

    return true;
}

if (count(debug_backtrace()) == 0) {
    TestRunner\pick("testLinearSearch");
}
